src/rum/instrumentation/DdRumErrorTracking.tsx

```diff
@@ -20,39 +21,87 @@ export class DdRumErrorTracking {
+
     /**
      * Starts tracking errors and sends a RUM Error event every time an error is detected.
      */
     static startTracking(): void {
         // extra safety to avoid wrapping the Error handler twice
-        if (this.isTracking) {
+        if (DdRumErrorTracking.isTracking) {
+            console.log("DdRumErrorTracking already started");
             return
         }
         
         if (ErrorUtils) {
-            this.defaultErrorHandler = ErrorUtils.getGlobalHandler();
+            DdRumErrorTracking.defaultErrorHandler = ErrorUtils.getGlobalHandler();
+            DdRumErrorTracking.defaultConsoleError = console.error;
 
-            ErrorUtils.setGlobalHandler(this.onError);
+            
+            ErrorUtils.setGlobalHandler(DdRumErrorTracking.onGlobalError);
+            console.error = DdRumErrorTracking.onConsoleError;
 
-            this.isTracking = true;
+            DdRumErrorTracking.isTracking = true;
         }
 
     }
 
     // eslint-disable-next-line 
-    static onError(error: any, isFatal?: boolean) {
-        // Stack trace
-        let stack = EMPTY_STACK_TRACE
-        if (error.hasOwnProperty('componentStack')) {
-            stack = String(error.componentStack);
-        } else if (error.hasOwnProperty('sourceURL') && error.hasOwnProperty('line') && error.hasOwnProperty('column')) {
-            stack = "at " + error.sourceURL + ":" + error.line + ":" + error.column
+    static onGlobalError(error: any, isFatal?: boolean) {
+        DdRum.addError(
+            String(error), 
+            TYPE_SOURCE, 
+            DdRumErrorTracking.getErrorStackTrace(error), 
+            new Date().getTime(), 
+            { "_dd.error.is_crash": isFatal, "_dd.error.raw": error }
+        ).then(() => {
+            DdRumErrorTracking.defaultErrorHandler(error, isFatal);
+        });
+    }
+
+    static onConsoleError(...params: unknown[]) {
+        let stack: string = EMPTY_STACK_TRACE
+        for (let i = 0; i < params.length; i += 1) {
+            const param = params[i];
+            const paramStack = DdRumErrorTracking.getErrorStackTrace(param)
+            if (paramStack != undefined && paramStack != EMPTY_STACK_TRACE){
+                stack = paramStack;
+                break;
+            }
         }
 
-        DdRum.addError(String(error), TYPE_SOURCE, stack, new Date().getTime(), error).then(() => {
-            this.defaultErrorHandler(error, isFatal)
+        const message = params.map((param) => {
+            if (typeof param === 'string') { return param }
+            else if (param instanceof Error) { return String(param) }
+            else { return JSON.stringify(param)}
+        }).join(' ');
+
+        
+        DdRum.addError(
+            message, 
+            TYPE_CONSOLE,
+            stack, 
+            new Date().getTime(), 
+            {}
+        ).then(() => {
+            DdRumErrorTracking.defaultConsoleError.apply(console, params);
```

Comment:
why here we are using `apply`, but in case of `DdRumErrorTracking.defaultErrorHandler(error, isFatal);` we don't?

Reply:
It is because of the  variable arguments `...params` , to ensure it's treated properly